script/update-content-versions-and-check-for-archived-slugs.ts

```diff
@@ -0,0 +1,110 @@
+import fs from 'fs';
+import { isEqual } from 'lodash';
+import path from 'path';
+import { argv } from 'yargs';
+import { content } from '../src/app/content/routes';
+import { LinkedArchiveTreeNode } from '../src/app/content/types';
+import { flattenArchiveTree } from '../src/app/content/utils';
+import { makeUnifiedBookLoader } from '../src/app/content/utils';
+import { findArchiveTreeNodeById } from '../src/app/content/utils/archiveTreeUtils';
+import { ARCHIVE_URL, REACT_APP_ARCHIVE_URL, REACT_APP_OS_WEB_API_URL } from '../src/config';
+import books from '../src/config.books';
+import createArchiveLoader from '../src/gateways/createArchiveLoader';
+import createOSWebLoader from '../src/gateways/createOSWebLoader';
+import { Redirects } from '../src/redirects/types';
+
+const args = argv as any as {
+  bookId: string
+  newVersion: string | number;
+};
+
+const booksPath = path.resolve(__dirname, '../src/config.books.json');
+const redirectsPath = path.resolve(__dirname, '../src/redirects/');
+
+const bookLoader = makeUnifiedBookLoader(
+  createArchiveLoader(`${ARCHIVE_URL}${REACT_APP_ARCHIVE_URL}`),
+  createOSWebLoader(`${ARCHIVE_URL}${REACT_APP_OS_WEB_API_URL}`)
+);
+
+async function updateRedirections(bookId: string, currentVersion: string, newVersion: string) {
+  if (currentVersion === newVersion) {
+    return 0;
+  }
+
+  const { tree: currentTree, slug: bookSlug } = await bookLoader(bookId, currentVersion)
+    .catch((error) => {
+      // tslint:disable-next-line: no-console
+      console.log(`error while loading book ${bookId} with defaultVersion ${currentVersion}`);
+      throw error;
+    });
+
+  const { tree: newTree } = await bookLoader(bookId, newVersion)
+    .catch((error) => {
+      // tslint:disable-next-line: no-console
+      console.log(`error while loading book ${bookId} with newVersion ${newVersion}`);
+      throw error;
+    });
+
+  const redirectsBookPath = path.resolve(redirectsPath, bookId + '.json');
+  const redirects: Redirects = fs.existsSync(redirectsBookPath) ? require(redirectsBookPath) : [];
```

Comment:
mix/matching import and require is going to cause problems someday, you may just want to change this to `await import` now